### ISSUE TITLE
All Scheduler Cards 'Could not connect with the scheduler component' in 2023.9

### ISSUE THREAD

**Issue by crlogic** [nielsfaber/scheduler-card#724]
### Checklist

- [X] I checked for similar existing issues (both open and closed) before posting.
- [X] I will participate in further discussion about this issue and can help by testing (if requested).

### Card Version

 v3.2.11

### Component Version

 v3.2.15

### Bug description

This is a long-standing install of over 1 year with 24 schedules. The schedules continue to work and I can turn them on/off in the Schedule integration.  It is the Scheduler Card (GUI) that only displays the following for all cards.

![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/cc1287e4-7906-42de-b059-ec37e2191650)

I have tried re-downloading both components from HACS and restarting HA.  I didn't see a debug config option nor log entries of value (to my limited knowledge).

### Steps to reproduce

Can't say exactly as it seemed to appear around the 2023.9 timeframe.  I don't change them much, mostly between seasons.

**Comment by nielsfaber** [comment]
I assume you tried clicking the refresh button or refresh the complete browser page?
Do you keep your HA installation up to date? Or could it be that the previous version of scheduler card was much (>6 months) older? In this case, I suggest to open the browser console to double-check which version of the card is actually running. It should be printed there with a badge.
Did you also try adding a new instance of a scheduler-card (with blank configuration)?

**Comment by crlogic** [comment]
Yes, sorry I didn't mention.  Refresh, CTRL+F5, Incognito, none resolve.

> Do you keep your HA installation up to date?

Affirm for core/supervisor/etc
![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/fd8d2cb4-dae9-47a2-9252-87afccc02ad9)

As for HACS, yes I typically update those roughly once a week.

> open the browser console to double-check which version of the card is actually running

Here you go

![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/53517610-4f7b-4f5c-8c6b-cb1ac396ad65)



> Did you also try adding a new instance of a scheduler-card (with blank configuration)?

This shows when "edit dashboard" is active and disappears when Done.
![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/40f20c31-897e-4c07-a4f3-39ce83d5b46b)

Added `grep` if helpful
![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/c2f9950d-9e55-44d3-b69c-3d8a1ea07954)

**Comment by nielsfaber** [comment]
Sadly to say, I am very much puzzled by this.
I am running exactly the same HA version + scheduler component/card versions, without any issue.
There is definitely not a compatibility issue nor a recent change that may have caused this problem (I would have also expected more users to raise the issue if this was the case).
The 'unknown command' messages which you are seeing in the log suggest that the scheduler-card is trying to make connection with the scheduler-component before scheduler-component has started.
This is not necessarily a problem, unless the startup of HA takes very long. However your logs show that scheduler is started within 10 seconds which I consider pretty normal. 
In summary, I don't have any ideas what's going on in your setup.
I

**Comment by crlogic** [comment]
Yeah I agree it is puzzling. It has worked so well for so long.

Is there perhaps a set of troubleshooting steps that might help?

If I remove all components and integrations will I lose all the configs?

**Comment by nielsfaber** [comment]
The schedules are stored in the .storage subfolder in the file scheduler.storage (in JSON format).
You could simply copy/backup this file to not risk losing your schedules, you can always place it back.
Obviously, this does not include the card configs, they are stored in the frontend (you could backup the yaml code).
The scheduler.storage file will be erased when you delete the scheduler integration.
But a fresh start may (miraculously) fix things, I cannot predict.

**Comment by crlogic** [comment]
Thanks for the tips.  I'll give that idea a go tomorrow and report back.

**Comment by crlogic** [comment]
Quick update that didn't work so well, perhaps the process needs review..

## Removal
1. `cp /root/config/.storage/scheduler.storage /root/config/.storage/scheduler.storage.bak`
2. `cp /root/config/.storage/lovelace.lovelace_schedule /root/config/.storage/lovelace.lovelace_schedule.bak`
3. Snapshot VM
4. Remove scheduler-card from HACS frontend
5. Delete Scheduler integration
6. Remove scheduler-component from HACS Integrations
7. Restart HA core

## Re-install
1. Install scheduler-component
2. `cp /root/config/.storage/scheduler.storage.bak /root/config/.storage/scheduler.storage`
3. Install scheduler-card
4. `cp /root/config/.storage/lovelace.lovelace_schedule.bak /root/config/.storage/lovelace.lovelace_schedule`
5. Restart HA core
6. Add Scheduler integration

After this, my Scheduler integration that previously had 24 entity had zero :(   So I reverted my snapshot.

Thoughts?

**Comment by crlogic** [comment]
Thanks @nielsfaber 

New re-install order
## Re-install
1. Install scheduler-card
2. Install scheduler-component
3. Restart HA core
4. Add Scheduler integration
5. `cp /root/config/.storage/scheduler.storage.bak /root/config/.storage/scheduler.storage`
6. `cp /root/config/.storage/lovelace.lovelace_schedule.bak /root/config/.storage/lovelace.lovelace_schedule`
7. Restart HA core

The scheduler-card's now show up, but the scheduler still did not retain the old entities so I reverted again.
![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/a7f1ed7f-60d1-4c4a-b4f0-5c924278c802)

Any other thoughts?

**Comment by nielsfaber** [comment]
This method has worked for me a dozen times, I don't know what you're doing different than me.
I also use it for investigating specific bugs, copying the scheduler storage file from a user and installing it in my local HA installation.

You may also try to create a single (random) schedule using the card.
Afterwards (10 seconds after creation) you can open the scheduler.storage and you will see the configuration here.
Then you could overwrite contents of this file with the contents of the older file and restart HA.
The result should be the same as your current approach, but a more cautious method.

**Comment by crlogic** [comment]
Is it possible that something is amiss with my scheduler.storage file?

**Comment by crlogic** [comment]
> You may also try to create a single (random) schedule using the card.
> Afterwards (10 seconds after creation) you can open the scheduler.storage and you will see the configuration here.
> Then you could overwrite contents of this file with the contents of the older file and restart HA.
> The result should be the same as your current approach, but a more cautious method.

I'll try this

**Comment by crlogic** [comment]
Results from partial restore

## Re-install
1. Install scheduler-component
2. Install scheduler-card
3. Restart HA core
4. Add Scheduler integration
5. Add a blank schedule and verify scheduler.storage
![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/74a4e6e7-a0dc-4215-b620-d42c93a77409)
7. Restart HA core
8. `cp /root/config/.storage/scheduler.storage.bak /root/config/.storage/scheduler.storage`
9. Confirm file contents successfully
10. Restart HA core
11. Check scheduler.storage and it is back to 
![image](https://github.com/nielsfaber/scheduler-card/assets/2413683/74a4e6e7-a0dc-4215-b620-d42c93a77409)

Gotta be missing something right!?

**Comment by crlogic** [comment]
Final working restore process

## Re-install
1. Install scheduler-component
2. Install scheduler-card
3. Restart HA core
4. Add Scheduler integration
5. Add a blank schedule and verify scheduler.storage
7. **STOP** HA core
8. `cp /root/config/.storage/scheduler.storage.bak /root/config/.storage/scheduler.storage`
10. Start HA core

**Comment by crlogic** [comment]
@nielsfaber - thanks for the guidance :)
